### PR TITLE
Make helpers work with dask

### PR DESCRIPTION
**Pull request by flying-sheep**
Preparation for more thorough testing of dask support

- [x] move `DaskArray` to `_compat` to have it available everywhere
- [x] test everything using `array_type` or similar with the same fixture
- [x] Add Dask support to `is_constant` (which uses above fixture)

the code in [scanpy/metrics/_common.py](https://github.com/scverse/scanpy/pull/2595/files#diff-caabb542dafdc95621693d71cb6a514af1457c05926438e307d3f8107bf91401) has been moved from [scanpy/metrics/_gearys_c.py](https://github.com/scverse/scanpy/pull/2595/files#diff-1ff58f43272e7b1451de7df9813a0d20aba57f55b23d38b2d46e309d75c2879b) without changes.

the only real meat this has is in [scanpy/_utils/compute/is_constant.py](https://github.com/scverse/scanpy/pull/2595/files#diff-e2c27335c5bfbbfb5ae9ac042c09411bfaca7d22340c257974d5d2bc68aea677):

https://github.com/scverse/scanpy/blob/f4f4185709b438e9cfe56db806a449634e214756/scanpy/_utils/compute/is_constant.py#L127-L134

it’s not a great implementation since it concatenates everything along an axis, but that makes it trivially correct.

### CODE REVIEW

File: scanpy/_utils/compute/is_constant.py [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+from numpy.typing import NDArray
 from numba import njit
 from scipy import sparse
 
+from ..._compat import DaskArray
+
+
+def _check_axis_supported(axis: Any) -> None:
+    if not isinstance(axis, Integral):
+        raise TypeError("axis must be integer or None.")
```
Comment (Zethson) [inline scanpy/_utils/compute/is_constant.py]: Just below you're testing whether the axis is 1 or 0.

Maybe simplify this by just raising a ValueError if the axis is not in `(None, 0, 1)`?

File: scanpy/tests/test_utils.py [hunk 2]
```diff
@@ -64,3 +62,32 @@ def test_is_constant(array_type):
...
+        pytest.param(0, [True, True, False, False], id='0'),
+        pytest.param(1, [False, False, True, True, False, True], id='1'),
+    ],
+)
+@pytest.mark.parametrize('block_type', [np.array, csr_matrix])
+def test_is_constant_dask(axis, expected, block_type):
+    import dask.array as da
+
+    if (axis is None) and block_type is csr_matrix:
+        pytest.skip('dask has weak suppoer for scipy sparse matrices')
```
Comment (Zethson) [inline scanpy/tests/test_utils.py]:
```suggestion
        pytest.skip('Dask has weak support for scipy sparse matrices')
```

**Review by Zethson — APPROVED** [review]
LGTM to my novice eyes

File: scanpy/_utils/compute/is_constant.py (hunk 1 above)
Comment (flying-sheep) [reply scanpy/_utils/compute/is_constant.py]: I clarified that different kinds of errors are supposed to be thrown and made sure the message matches the check.

**Pull request merged**